Dataset correction and improvements

Enmartz opened the pull request.

leonsuarez24 commented:
> _ No description provided. _

Enmartz commented:
> > _ No description provided. _
> 
> You didn't provide a description

This issue is related to issue #173, as mentioned in the conversation history. Please check it out to review the changes and improvements I made.

leonsuarez24 reviewed (changes requested):
Thanks @Enmartz , the datasets are working perfectly.


To further improve the Datasets Module, I suggest updating the documentation to clearly state that the train dataset can be accessed using the train keyword argument. By default, the module uses the validation set for training, and users would need to examine the code to understand how to switch to the training set. Providing this information in both the function documentation and the documentation page would prevent users from having to look into the code.

![image](https://github.com/user-attachments/assets/36312da2-14fa-47ee-9f8c-4bfc72d07dba)

Enmartz commented:
> Thanks @Enmartz , the datasets are working perfectly.
> 
> To further improve the Datasets Module, I suggest updating the documentation to clearly state that the train dataset can be accessed using the train keyword argument. By default, the module uses the validation set for training, and users would need to examine the code to understand how to switch to the training set. Providing this information in both the function documentation and the documentation page would prevent users from having to look into the code.
> 
> ![image](https://private-user-images.githubusercontent.com/138550157/416359349-36312da2-14fa-47ee-9f8c-4bfc72d07dba.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDEwMTkxMTgsIm5iZiI6MTc0MTAxODgxOCwicGF0aCI6Ii8xMzg1NTAxNTcvNDE2MzU5MzQ5LTM2MzEyZGEyLTE0ZmEtNDdlZS05ZjhjLTRiZmM3MmQwN2RiYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMwM1QxNjIwMThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZTA1ODQwMTU1ZWUwMDEzMWViOGEzYzVjMDcwMjgxZDc4ZmU3MDJmZDFlOTllOWFmOGE3Zjg1YWQyNmRmM2U0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.3iyjBoftncqfcEn91IpRc5g9I5mhqWndyo3GCbl8IyI)

Thanks for the improvement advice. I implemented it by removing `**kwargs` and explicitly defining the parameters `train` and `download`. I also modified the rest of the necessary code to adapt this improvement. Please check it out.

leonsuarez24 reviewed (approved):
LGTM
